http/src/request/mod.rs

```diff
@@ -93,15 +93,54 @@ use crate::{
 use bytes::Bytes;
 use hyper::{
     header::{HeaderMap, HeaderName, HeaderValue},
-    Method,
+    Method as HyperMethod,
 };
 use percent_encoding::{utf8_percent_encode, NON_ALPHANUMERIC};
-
 use std::{borrow::Cow, future::Future, pin::Pin};
 
 type Pending<'a, T> = Pin<Box<dyn Future<Output = Result<T>> + Send + 'a>>;
 type PendingOption<'a> = Pin<Box<dyn Future<Output = Result<Bytes>> + Send + 'a>>;
 
+/// Request method.
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+#[non_exhaustive]
+pub enum Method {
```

Comment:
Is there a good reason to include all of these options when only GET, DELETE, PATCH, POST, and PUT are used? The conversion to Hyper's method type is one way, so the only way those variants will be used is via raw requests.

It might make sense when making raw requests when twilight has not been updated, but I don't think Discord is going to make a public endpoint that requires any of the other methods listed here in the foreseeable future.

Reply:
Addressed